File: library/lib_module_http/src/main/java/org/cchao/http/CacheConsumer.java
package org.cchao.http;

import org.cchao.common.utils.JsonUtils;
import org.cchao.common.utils.Md5Utils;
import org.cchao.http.db.CacheDbUtils;
import org.cchao.http.db.CacheModel;

import io.reactivex.functions.Consumer;

/**
 * @author cchen6
 * @Date on 2019/8/5
 * @Description
 */
public class CacheConsumer implements Consumer<HttpResponseModel<Object>> {

    private HttpRequestBody httpRequestBody;

    private boolean isCache;

    public CacheConsumer(HttpRequestBody httpRequestBody, boolean isCache) {
        this.httpRequestBody = httpRequestBody;
        this.isCache = isCache;
    }

    @Override
    public void accept(HttpResponseModel<Object> objectHttpResponseModel) {
        if (!isCache || null == objectHttpResponseModel || !objectHttpResponseModel.isSuccess()) {
            return;
        }
        String key = Md5Utils.getMd5(JsonUtils.toString(httpRequestBody));
        CacheModel cacheModel = CacheDbUtils.getInstance().queryCacheModel(key);
        if (null == cacheModel) {
            cacheModel = new CacheModel(key, JsonUtils.toString(objectHttpResponseModel.getData()), objectHttpResponseModel.getCode(), objectHttpResponseModel.getMsg(), System.currentTimeMillis());
            CacheDbUtils.getInstance().saveCache(cacheModel);
        } else {
            cacheModel.setTime(System.currentTimeMillis());
            cacheModel.setContent(JsonUtils.toString(objectHttpResponseModel.getData()));
            cacheModel.setCode(objectHttpResponseModel.getCode());
            cacheModel.setMsg(objectHttpResponseModel.getMsg());
        }
    }
}

File: library/lib_module_http/src/main/java/org/cchao/http/CacheConsumer.kt
package org.cchao.http

import io.reactivex.functions.Consumer
import org.cchao.common.utils.JsonUtils
import org.cchao.common.utils.Md5Utils
import org.cchao.http.db.CacheDbUtils
import org.cchao.http.db.CacheModel

/**
 * @author cchen6
 * @Date on 2019/8/5
 * @Description
 */
internal class CacheConsumer(private val httpRequestBody: HttpRequestBody, private val isCache: Boolean) : Consumer<HttpResponseModel<Any>> {

    override fun accept(objectHttpResponseModel: HttpResponseModel<Any>?) {
        if (!isCache || null == objectHttpResponseModel || !objectHttpResponseModel.isSuccess) {
            return
        }
        val key = Md5Utils.getMd5(JsonUtils.toString(httpRequestBody))
        var cacheModel = CacheDbUtils.instance.queryCacheModel(key)
        if (null == cacheModel) {
            cacheModel = CacheModel(key, JsonUtils.toString(objectHttpResponseModel.data), objectHttpResponseModel.code, objectHttpResponseModel.msg, System.currentTimeMillis())
            CacheDbUtils.instance.saveCache(cacheModel)
        } else {
            cacheModel.time = System.currentTimeMillis()
            cacheModel.content = JsonUtils.toString(objectHttpResponseModel.data)
            cacheModel.code = objectHttpResponseModel.code
            cacheModel.msg = objectHttpResponseModel.msg
        }
    }
}
